code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -762,6 +762,111 @@
+	if(!spawn_locs.len || spawn_locs.len < 15) //look for any morgue trays, crematoriums, ect if there weren't alot of dead bodies on the station to pick from
+		for(var/obj/structure/bodycontainer/bc in GLOB.bodycontainers)
+			var/turf/T = get_turf(bc)
+			if(T && is_station_level(T.z))
+				spawn_locs += T
+	if(!spawn_locs.len) //If we can't find any valid spawnpoints, try the carp spawns
+		for(var/obj/effect/landmark/carpspawn/L in GLOB.landmarks_list)
+			if(isturf(L.loc))
+				spawn_locs += L.loc
+	if(!spawn_locs.len) //If we can't find THAT, then just give up and cry
+		return FALSE
+	. = ..()
+
+/datum/dynamic_ruleset/midround/from_ghosts/revenant/generate_ruleset_body(mob/applicant)
+	var/mob/living/simple_animal/revenant/revvie = new(pick(spawn_locs))
+	revvie.key = applicant.key
+	message_admins("[ADMIN_LOOKUPFLW(revvie)] has been made into a revenant by the midround ruleset.")
+	log_game("[key_name(revvie)] was spawned as a revenant by the midround ruleset.")
+	return revvie
+
+//////////////////////////////////////////////
+//                                          //
+//       SENTIENT DISEASE (GHOST)           //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/midround/from_ghosts/sentient_disease
+	name = "Sentient Disease"
+	antag_datum = /datum/antagonist/disease
+	antag_flag = "Sentient Disease"
+	antag_flag_override = ROLE_ALIEN
+	required_candidates = 1
+	weight = 4
+	cost = 10
+	requirements = list(101,101,101,80,60,50,30,20,10,10)
+	repeatable = TRUE
+	var/list/spawn_locs = list()
+
+/datum/dynamic_ruleset/midround/from_ghosts/sentient_disease/generate_ruleset_body(mob/applicant)
+	var/mob/camera/disease/virus = new /mob/camera/disease(SSmapping.get_station_center())
+	virus.key = applicant.key
+	INVOKE_ASYNC(virus, /mob/camera/disease/proc/pick_name)
+	message_admins("[ADMIN_LOOKUPFLW(virus)] has been made into a sentient disease by the midround ruleset.")
+	log_game("[key_name(virus)] was spawned as a sentient disease by the midround ruleset.")
+	return virus
+
+//////////////////////////////////////////////
+//                                          //
+//       SPACE PIRATES     (GHOST)          //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/midround/pirates
+	name = "Space Pirates"
+	antag_flag = "Space Pirates"
+	required_type = /mob/dead/observer
+	enemy_roles = list("Security Officer", "Detective", "Head of Security", "Captain")
+	required_enemies = list(2,2,1,1,1,1,1,0,0,0)
+	required_candidates = 3
+	weight = 4
+	cost = 10
+	requirements = list(101,101,101,80,60,50,30,20,10,10)
+	repeatable = TRUE
+
+/datum/dynamic_ruleset/midround/pirates/execute()
+	//A lot of pirate code is baked into the event directly, so let's just make a copy of the event and opt to run it.
+	var/datum/round_event_control/pirates/pirate_event = new/datum/round_event_control/pirates
```

Comment:
You don't want this.

Otherwise, stuff like getting no votes spawning a sleeper agent instead won't work.

Abstract this to a proc/datum/whatever the hell and have them both use it. Make sure that `pirates` is a ghost ruleset just like any other.